module/mempool/queue/entityRequests.go

```diff
@@ -0,0 +1,103 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/mempool"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+	"github.com/onflow/flow-go/module/mempool/queue/internal"
+)
+
+// EntityRequestStore is a FIFO (first-in-first-out) size-bound queue for maintaining EntityRequests.
+// It is designed to be utilized at the common ProviderEngine to maintain and respond entity requests.
```

Comment:
```suggestion
// It is designed to be utilized at the common ProviderEngine to maintain and respond to entity requests.
```

Reply:
`EntityRequests` has totally removed and generalized as `HeroStore`.